src/Microsoft.Diagnostics.Runtime/ClrModule.cs

```diff
@@ -31,26 +31,49 @@ public sealed class ClrModule : IClrModule
         private ulong? _size;
         private ClrHeap? _heap;
 
-        internal ClrModuleInfo ModuleInfo { get; }
+        private int initialized;
+        private readonly ulong _moduleAddress;
+        private ClrModuleInfo _moduleInfo;
+
+        internal ClrModuleInfo ModuleInfo
+        {
+            get
+            {
+                if (initialized == 1)
```

Comment:
```suggestion
                if (Volatile.Read(ref initialized) == 1)
```
If the variable itself isn't marked volatile all reads and writes must be interlocked or volatile helper methods in order to be properly treated by the JIT/CPU